ggml/src/ggml-webgpu/wgsl-shaders/cpy.wgsl

```diff
@@ -0,0 +1,60 @@
+enable f16;
+
+@group(0) @binding(0)
+var<storage, read_write> src: array<f32>;
+
+@group(0) @binding(1)
+var<storage, read_write> dst: array<f16>;
+
+struct Params {
+    ne: u32,            // total number of elements
+    offset_src: u32,    // in bytes
+    offset_dst: u32,    // in bytes
+
+    // Strides (in elements) — may be permuted
+    stride_src0: u32,
+    stride_src1: u32,
+    stride_src2: u32,
+    stride_src3: u32,
+
+    stride_dst0: u32,
+    stride_dst1: u32,
+    stride_dst2: u32,
+    stride_dst3: u32,
```

Comment:
I think I remember a comment about WebGPU not supporting 64-bit integers. Is this correct?

If it cannot support them, then it might get quite tricky, especially with the new `ggml_set_rows()` requiring I64 data (#14274).

Reply:
Yes, WebGPU currently does not support 64-bit integers. However, it is at least on their radar to support it, as well as other data types: https://github.com/gpuweb/gpuweb/issues/273

Do you think this is something that could be emulated for now, or is it a showstopper, or just something we should be aware of for now?

Reply:
Not a showstopper. We just have to be careful when implementing `GGML_OP_SET_ROWS` to take the lower 32-bits of the indices and if possible to raise an error when the upper bits are non-zero.